tests/unit/transport/test_transport_authz_handling.py

```diff
@@ -28,3 +28,27 @@ def test_will_null_authz_header_with_null_authorizer():
     request.headers["Authorization"] = "foo bar"
     transport._set_authz_header(NullAuthorizer(), request)
     assert request.headers == {}
+
+
+def test_request_caller_info_creation():
+    mock_authorizer = mock.Mock()
+    caller_info = RequestCallerInfo(authorizer=mock_authorizer)
+
+    assert caller_info.authorizer is mock_authorizer
+
+
+def test_requests_transport_accepts_caller_info():
+    transport = RequestsTransport()
+    mock_authorizer = mock.Mock()
+    mock_authorizer.get_authorization_header.return_value = "Bearer token"
+    caller_info = RequestCallerInfo(authorizer=mock_authorizer)
+
+    with mock.patch.object(transport, "session") as mock_session:
+        mock_response = mock.Mock(status_code=200)
+        mock_session.send.return_value = mock_response
+
+        response = transport.request(
+            "GET", "https://example.com", caller_info=caller_info
+        )
+
+        assert response.status_code == 200
```

Comment:
I think I'd like to see the headers on the sent request here, to confirm that `headers={"Authorization": "Bearer token"}` -- meaning that the "unwrapping" of this data was correct.